Sources/Build/SPMSwiftDriverExecutor.swift

```diff
@@ -0,0 +1,75 @@
+/*
+ This source file is part of the Swift.org open source project
+
+ Copyright 2020 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See http://swift.org/LICENSE.txt for license information
+ See http://swift.org/CONTRIBUTORS.txt for Swift project authors
+*/
+
+@_implementationOnly import SwiftDriver
+import TSCBasic
+import Foundation
+
+final class SPMSwiftDriverExecutor: DriverExecutor {
+  let resolver: ArgsResolver
+  let fileSystem: FileSystem
+  let env: [String: String]
+
+  init(resolver: ArgsResolver,
+       fileSystem: FileSystem,
+       env: [String: String]) {
+    self.resolver = resolver
+    self.fileSystem = fileSystem
+    self.env = env
+  }
+
+  func execute(job: Job,
+               forceResponseFiles: Bool,
+               recordedInputModificationDates: [TypedVirtualPath : Date]) throws -> ProcessResult {
+    let arguments: [String] = try resolver.resolveArgumentList(for: job,
+                                                               forceResponseFiles: forceResponseFiles)
+
+    try job.verifyInputsNotModified(since: recordedInputModificationDates,
+                                    fileSystem: fileSystem)
+    assert(!job.requiresInPlaceExecution,
```

Comment:
I don't see how this can fail *now*, but it seems fragile. Should we throw an error here rather than crash, given the likelihood that someone can slip an argument through to the Swift driver that would cause in-place execution? Crashing SwiftPM seems unfortunate there.

Reply:
Good idea! I changed this to throw an error instead